roles/kubernetes/control-plane/tasks/kubeadm-setup.yml

```diff
@@ -81,12 +81,21 @@
     mode: 0640
 
 - name: kubeadm | Check if apiserver.crt contains all needed SANs
-  command: openssl x509 -noout -in "{{ kube_cert_dir }}/apiserver.crt" -check{{ item|ipaddr|ternary('ip','host') }} "{{ item }}"
-  with_items: "{{ apiserver_sans }}"
+  shell: |
+    set -o pipefail
+    {% set apiserver_ips = apiserver_sans|map('ipaddr')|reject('equalto', False)|list %}
```

Comment:
This would be more readable if you moved these to vars section like this:
```
- name: task name
  shell: |
    for IP in ...
      some logic
      echo 'NEED-RENEW'
    done
  args:
    executable: /bin/bash
  vars:
    apiserver_ips: "{{ apiserver_sans|map('ipaddr')|reject('equalto', False)|list }}"
     apiserver_hosts: "{{ apiserver_sans|difference(apiserver_ips) }}"
  register: apiserver_sans_check
  changed_when: ...
  when: ...
Finally, why do we need executable: /bin/bash? There's no bashisms in this and sh is adequate here.

Reply:
pipefail is a bashisms, but I'll definitely move the vars

Reply:
vars moved, please recheck